src/compute_graph/tree.jl

```diff
@@ -0,0 +1,87 @@
+using AbstractTrees
+using YAML
+
+export build_tree, ComputeNode, ComputeTree, get_commands
+# In this file we define a tree data structure which can be used for optimisation passes
+# over contraction commands
+
+"""Generic binary tree node data structure"""
```

Comment:
```suggestion
"""Generic tree node data structure"""
```

I guess technically this won't be a binary tree anymore